lib/identity/realm.js

```diff
@@ -250,6 +251,38 @@ class Realm extends CryptoConsumer {
     return idClient
   }
 
+  async registerFederatedUser(username, data, cryptoKeys, signingKeys) {
+    // eslint-disable-next-line
+    console.log(username, data) // flag if seen in CR
+    const { result } = data.context
+    try {
+      const user = await this.fromObject({
+        config: {
+          api_url: this.apiUrl,
+          realm_name: this.realmName,
+          username: username,
+          user_id: result.identity.id,
+        },
+        storage: {
+          version: 2,
+          client_id: result.identity.tozny_id,
+          api_key_id: result.identity.api_key_id,
+          api_secret: result.identity.api_secret_key,
+          public_key: cryptoKeys.publicKey,
+          private_key: cryptoKeys.privateKey,
+          public_signing_key: signingKeys.publicKey,
+          private_signing_key: signingKeys.privateKey,
+          api_url: this.apiUrl,
+        },
+      })
+      const brokerId = result.realm_broker_identity_tozny_id
+      await writeFederationNote(user, brokerId, true)
+    } catch (e) {
+      // eslint-disable-next-line
+      console.error(e) // flag if seen in CR
```

Comment:
same as above, we probably want to do more than console log if we're handling the error